src/handlers/asg/event.py

```diff
@@ -101,6 +106,15 @@ def handle_ha_event(client, lambda_client, controller_instanceobj, context):
         if is_upgrade_to_build_supported(controller_api_ip, cid):
             ctrl_version = ctrl_version_with_build
 
+        if os.path.exists(DEV_FLAG):
+            if enable_central_services_staging_mode(cid, controller_api_ip):
```

Comment:
optional:
You could combine these two statements but it only removes some of the nested depth not a big deal:

```if os.path.exists(DEV_FLAG) and enable_central_services_staging_mode(cid, controller_api_ip):```